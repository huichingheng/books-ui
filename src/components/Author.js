import React, { Component } from "react";

const API_HOST = process.env.REACT_APP_BOOKS_API || "http://localhost:3000";

class Author extends Component {
  constructor() {
    super();
    this.state = {
      authors: []
    };
  }

  async getAuthor() {
    const url = `${API_HOST}/authors`;
    const response = await fetch(url);
    const data = await response.json();
    this.setState({
      authors: data
    });
  }

  async componentDidMount() {
    await this.getAuthor();
  }
  render() {
    return (
      <div>
        <h2>Authors</h2>
        {this.state.authors.map(author => {
          return <li key={author._id}>{author.name}</li>;
        })}
      </div>
    );
  }
}

export default Author;
