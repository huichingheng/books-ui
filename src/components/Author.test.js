import React from "react";
import ReactDOM from "react-dom";
import Author from "./Author";
import ShallowRender from "react-test-renderer/shallow";

test("renders Author component", () => {
  const renderer = new ShallowRender();
  renderer.render(<Author />);
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});