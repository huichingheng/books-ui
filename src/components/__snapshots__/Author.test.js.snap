// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Author component 1`] = `
<div>
  <h2>
    Authors
  </h2>
</div>
`;

exports[`renders Book component 1`] = `
<div>
  <h2>
    Books
  </h2>
  ;
</div>
`;
